### PR TITLE
Add resources page

### PR DESCRIPTION
**Pull request by Lxkas**

### CODE REVIEW

**Comment by Lxkas** [comment]
@nunmanasm 

https://github.com/stamford-syntax-club/stamfordcenter-frontend/assets/24469918/6d1ba035-148b-47b9-9602-85c9d22fb645

**Comment by chinathaip** [comment]
me having OCD:

noooooooooooooo don't merge, rebase onto mainnnn instead ;3;
image

**Comment by Lxkas** [comment]
> me having OCD:
> 
> noooooooooooooo don't merge, rebase onto mainnnn instead ;3; image

You do it XD

File: src/pages/resources.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { useDebouncedValue, useDisclosure } from "@mantine/hooks";
+import { Resource } from "@ourtypes/resources_page_types/ResourceTypes";
+import { useEffect, useState } from "react";
+
+export default function ResourcesPage() {
+	const [resourceList, setResourceList] = useState<Resource[]>([]);
+	const [searchQuery, setSearchQuery] = useState("");
+
+	useEffect(() => {
+		fetch("https://center-be.stamford.dev/api/resources/resources").then((res) => {
```
Comment (chinathaip) [inline src/pages/resources.tsx]: https://github.com/stamford-syntax-club/stamfordcenter-frontend/issues/35
XDDDD

Comment (chinathaip) [reply src/pages/resources.tsx]: btw why don't you use SWR just like in `study_plans.tsx`?

**Review by chinathaip — CHANGES_REQUESTED** [review]
can you also add this page to the `navigationItems`? 

and rebase onto main too :3

**Comment by Lxkas** [comment]
> can you also add this page to the `navigationItems`?
> 
> and rebase onto main too :3

What page? XD

File: src/pages/resources.tsx (hunk 1 above)
Comment (Lxkas) [reply src/pages/resources.tsx]: ‘Cuz it doesn’t need to be updated constantly, swr is Stale While Revalidate, we don’t need to keep pulling in new data kekw

**Comment by chinathaip** [comment]
> > can you also add this page to the `navigationItems`?
> 
> > 
> 
> > and rebase onto main too :3
> 
> 
> 
> What page? XD

Resources page

**Comment by Lxkas** [comment]
> > > can you also add this page to the `navigationItems`?
> > 
> > 
> > > 
> > 
> > 
> > > and rebase onto main too :3
> > 
> > 
> > What page? XD
> 
> Resources page

OK

File: src/pages/resources.tsx (hunk 1 above)
Comment (chinathaip) [reply src/pages/resources.tsx]: not resolved!

DYNAMIC HOSTTTT

I want to see this in https://center-beta.stamford.dev
```
"name": "Class Schedules & Full Name Lists Beta Env งับ คริๆ",
```

and this in https://center.stamford.dev (except I haven't made the production deployment for you guys)
```
"name": "Class Schedules & Full Name Lists",
```

File: src/pages/study_plans.tsx [hunk 2]
```diff
@@ -19,7 +20,7 @@ async function fetcher(url: string): Promise<StudyPlanElement[]> {
 }
 
 export default function StudyPlansPage() {
-	const { data, error, isLoading } = useSWR("https://center-be.stamford.dev/api/study_plans", fetcher);
+	const { data, error, isLoading } = useSWR(`${API_BASE_LINK}/api/study_plans`, fetcher);
```
Comment (chinathaip) [inline src/pages/study_plans.tsx]: One thing Mr Tawan, I'm gonna stop supporting this endpoint now, can you migrate to the new one?

`/api/resources/study_plans`

https://github.com/stamford-syntax-club/stamfordcenter-backend/issues/15
image

**Comment by chinathaip** [comment]
closes https://github.com/stamford-syntax-club/stamfordcenter-frontend/issues/35
closes https://github.com/stamford-syntax-club/stamfordcenter-frontend/issues/25

**Review by chinathaip — APPROVED** [review]
LGTM

**Pull request merged**